src/toast/toast.test.tsx

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import {render, screen} from "@testing-library/react";
+import {waitFor} from "@testing-library/dom";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+
+import {Button, Toast, ToastContextProvider, useToaster} from "..";
+import {ToastContextState} from "./util/toastTypes";
+import {DEFAULT_TOAST_TIMEOUT} from "./util/toastConstants";
+
+const testToastContentText = "This is a test toast";
+
+const testToastData: ToastContextState["toastStack"][0] = {
+  id: "TestToast",
+  render() {
+    return <p>{testToastContentText}</p>;
+  }
+};
+
+const displayToastButtonText = "Display Toast";
+const hideToastButtonText = "Hide Toast";
+
+function ButtonWithToast({
+  numberOfToasts = 1,
+  autoClose,
+  timeout,
+  renderCustomToast
+}: {
+  numberOfToasts?: number;
+  autoClose?: boolean;
+  timeout?: number;
+  renderCustomToast?: () => React.ReactNode;
+}) {
+  const toast = useToaster();
+
+  return (
+    <>
+      <Button onClick={displayToasts}>{displayToastButtonText}</Button>
+      <Button onClick={hideToast}>{hideToastButtonText}</Button>
+    </>
+  );
+
+  function displayToasts() {
+    for (let i = 0; i < numberOfToasts; i++) {
+      toast.display({
+        ...testToastData,
+        id: `${testToastData.id}-${i + 1}`,
+        autoClose,
+        timeout,
+        render: renderCustomToast || testToastData.render
+      });
+    }
+  }
+
+  function hideToast() {
+    toast.hide(`${testToastData.id}-1`);
+  }
+}
+
+describe("<ToastContextProvider />", () => {
+  it("displays a toast that closes itself in default timeout", async () => {
+    const wrapper = ({children}: {children: React.ReactNode}) => (
+      <ToastContextProvider>{children}</ToastContextProvider>
+    );
+
+    render(<ButtonWithToast />, {wrapper});
+
+    const button = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(button);
+
+    expect(await screen.findByText(testToastContentText)).toBeVisible();
+
+    await waitFor(
+      () => {
+        expect(screen.queryByText(testToastContentText)).not.toBeInTheDocument();
+      },
+      {timeout: DEFAULT_TOAST_TIMEOUT}
```

Comment:
Is this working as expected? Can there be some kind of race condition, because we are closing the modal and checking if it was closed at the same time.